tests/unit/test_webmcp_variants.py

```
@@ -0,0 +1,228 @@
+    naive_tools = tool_names("naive", monkeypatch=monkeypatch)
+    assert "webmcp_call" in naive_tools
+    assert {"click", "type", "answer"}.issubset(naive_tools)
+
+    monkeypatch.setenv("CUGA_WEBMCP_MODE", "advanced")
+    assert tool_names("advanced", "tool_stage") == {"answer", "observe_page", "webmcp_call"}
+    assert "webmcp_call" not in tool_names("advanced", "page_fallback")
+    assert {"click", "type", "answer"}.issubset(tool_names("advanced", "page_fallback"))
+
+
+def test_webmcp_call_requires_page_config():
+    with pytest.raises(ValueError, match="configurable.page"):
```

Ruff (0.15.15)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)